editor/plugins/visual_shader_editor_plugin.cpp

```diff
@@ -104,6 +105,94 @@ void VisualShaderNodePlugin::_bind_methods() {
+		port_icon = get_theme_icon("port", "GraphNode");
+	}
+
+	icon_offset = -port_icon->get_size() * 0.5;
+
+	// Draw "shadow"/outline in the connection rim color.
+	draw_texture_rect(port_icon, Rect2(p_pos + icon_offset - Size2(2, 2), port_icon->get_size() + Size2(4, 4)), false, p_rim_color);
+	draw_texture(port_icon, p_pos + icon_offset, p_color);
+}
+
+void VSGraphNode::draw_port(int p_slot_index, Point2i p_pos, bool p_left, const Color &p_color) {
+	Color rim_color = get_theme_color("connection_rim_color", "GraphEdit");
+	_draw_port(p_slot_index, p_pos, p_left, p_color, rim_color);
+}
+
+///////////////////
+
+void VSRerouteNode::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("set_icon_opacity", "opacity"), &VSRerouteNode::set_icon_opacity);
+	ClassDB::bind_method(D_METHOD("get_icon_opacity"), &VSRerouteNode::get_icon_opacity);
+
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "icon_opacity", PROPERTY_HINT_NONE, "", PROPERTY_USAGE_NO_EDITOR | PROPERTY_USAGE_INTERNAL), "set_icon_opacity", "get_icon_opacity");
+}
+
+void VSRerouteNode::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_READY: {
+			connect("mouse_entered", callable_mp(this, &VSRerouteNode::_on_mouse_entered));
+			connect("mouse_exited", callable_mp(this, &VSRerouteNode::_on_mouse_exited));
+		} break;
+		case NOTIFICATION_DRAW: {
+			Vector2 offset = Vector2(0, -16);
+			Color drag_bg_color = get_theme_color("drag_background", "VSRerouteNode");
+			draw_circle(get_size() * 0.5 + offset, 16, Color(drag_bg_color, selected ? 1 : icon_opacity));
+
+			Ref<Texture2D> icon = get_theme_icon("ToolMove", "EditorIcons");
+			Point2 icon_offset = -icon->get_size() * 0.5 + get_size() * 0.5 + offset;
+			draw_texture(icon, icon_offset, Color(1, 1, 1, selected ? 1 : icon_opacity));
+		} break;
+	}
+}
+
+void VSRerouteNode::draw_port(int p_slot_index, Point2i p_pos, bool p_left, const Color &p_color) {
+	Color rim_color = selected ? get_theme_color("selected_rim_color", "VSRerouteNode") : get_theme_color("connection_rim_color", "GraphEdit");
+	_draw_port(p_slot_index, p_pos, p_left, p_color, rim_color);
+}
+
+VSRerouteNode::VSRerouteNode() {
+	Label *title_lbl = Object::cast_to<Label>(get_titlebar_hbox()->get_child(0));
+	title_lbl->hide();
+
+	const Size2 size = Size2(32, 32);
+
+	Control *slot_area = memnew(Control);
+	slot_area->set_custom_minimum_size(size);
+	slot_area->set_mouse_filter(Control::MOUSE_FILTER_IGNORE);
+	add_child(slot_area);
+
+	// Lay the input and output ports on top of each other to create the illusion of a single port.
+	add_theme_constant_override("port_h_offset", size.width / 2);
+}
+
+void VSRerouteNode::set_icon_opacity(float p_opacity) {
+	icon_opacity = p_opacity;
+	queue_redraw();
+}
+
+float VSRerouteNode::get_icon_opacity() const {
+	return icon_opacity;
+}
+
+void VSRerouteNode::_on_mouse_entered() {
+	Ref<Tween> tween = create_tween();
+	tween->tween_property(this, NodePath("icon_opacity"), 1, ANIMATION_LENGTH_SEC);
```

Comment:
You can avoid adding a property if you use `tween_method()` with `callable_mp()` on the setter.

Reply:
Try with `0.0` and `1.0`.

Reply:
Oh, I see. Yeah, thanks, it works now :)